pilot/pkg/model/service.go

```diff
@@ -521,6 +521,39 @@ func (ep *IstioEndpoint) IsDiscoverableFromProxy(p *Proxy) bool {
 	return ep.DiscoverabilityPolicy.IsDiscoverableFromProxy(ep, p)
 }
 
+// Metadata returns the endpoint metadata used for telemetry purposes.
+func (ep *IstioEndpoint) Metadata() *EndpointMetadata {
+	return &EndpointMetadata{
```

Comment:
I think embedding would look cleaner.

Reply:
I agree. That was my original change. But it resulted in too many changes - just worried about regression. I can change  like that. WDYT?

Reply:
No strong opinion, just looked like it makes sense here but It's possible it doesn't work elsewhere.